compiler/rustc_codegen_ssa/src/back/rpath.rs

```diff
@@ -82,12 +80,11 @@ fn get_rpath_relative_to_output(config: &mut RPathConfig<'_>, lib: &Path) -> OsS
     // Mac doesn't appear to support $ORIGIN
     let prefix = if config.is_like_osx { "@loader_path" } else { "$ORIGIN" };
 
-    let cwd = env::current_dir().unwrap();
-    let mut lib = fs::canonicalize(&cwd.join(lib)).unwrap_or_else(|_| cwd.join(lib));
-    lib.pop(); // strip filename
-    let mut output = cwd.join(&config.out_filename);
-    output.pop(); // strip filename
-    let output = fs::canonicalize(&output).unwrap_or(output);
+    // Strip filenames
+    let lib = lib.parent().unwrap();
+    let output = config.out_filename.parent().unwrap();
+    let lib = lib.canonicalize().unwrap();
+    let output = output.canonicalize().unwrap();
```

Comment:
Canonicalization can fail and on Windows there is a ramdisk implementation where it always fails.

Reply:
We now have `try_canonicalize` in [`rustc_fs_util`](https://github.com/rust-lang/rust/blob/d4ba2b4c7c938cf90c03a265cb31356537f608ad/compiler/rustc_fs_util/src/lib.rs#L97). It tries to get the canonical path and falls back to merely getting an absolute path.

Reply:
Nice! Done.